File: packages/types/src/core/system.ts
/**
 * 磁盘状态
 * @param mountpoint  string  挂载点
 * @param device  string  设备
 * @param fstype  string  文件系统类型
 * @param total  number  总空间
 * @param used  number  已用空间
 * @param free  number  空闲空间
 * @param percent  number  使用百分比
 */
export interface DiskStatus {
  mountpoint: string;
  device: string;
  fstype: string;
  total: number;
  used: number;
  free: number;
  percent: number;
}
/**
 * 系统信息
 * @param cpu_percent  number  cpu使用率
 * @param memory_total  number  内存总大小
 * @param memory_used  number  内存使用大小
 * @param swap_total  number  交换总大小
 * @param swap_used  number  交换使用大小
 * @param disks  DiskStatus[]  磁盘状态列表
 */
export interface SystemInfo {
  cpu_percent: number;
  memory_total: number;
  memory_used: number;
  swap_total: number;
  swap_used: number;
  disks: DiskStatus[];
}

File: ktorBackend/src/main/kotlin/com/qianrenni/models/domain/system.kt
package com.qianrenni.models.domain

import kotlinx.serialization.Serializable

/**
 * 单个磁盘分区的状态
 * 对应 Python 版 DiskStatus 模型
 */
@Serializable
data class DiskStatus(
    val mountPoint: String,
    val device: String,
    val fStype: String,
    val total: Long,
    val used: Long,
    val free: Long,
    val percent: Double,
)

/**
 * 系统状态
 * 对应 Python 版 SystemStatus 模型
 */
@Serializable
data class SystemStatus(
    val cpuPercent: Double,
    val memoryTotal: Long,
    val memoryUsed: Long,
    val swapTotal: Long,
    val swapUsed: Long,
    val disks: List<DiskStatus>,
)
